plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcPageSink.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.RowPagesBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.trino.RowPagesBuilder.rowPagesBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestJdbcPageSink
+{
+    private static final String TABLE_SQL = "CREATE TABLE example.dummy (txt varchar(36), num bigint)";
+    private static final String INSERT_SQL = "INSERT INTO example.dummy(txt, num) VALUES (?, ?)";
+    private static final String COUNT_SQL = "SELECT COUNT(*) FROM example.dummy";
+    private static final List<Type> COLUMN_TYPES = Arrays.asList(VarcharType.VARCHAR, BigintType.BIGINT);
+    private static final List<WriteFunction> COLUMN_WRITERS = Arrays.asList(
+            (SliceWriteFunction) (statement, index, value) -> statement.setString(index, value.toStringUtf8()),
+            (LongWriteFunction) PreparedStatement::setLong);
+
+    private TestingDatabase db;
+    private Connection connection;
+    private PreparedStatement statement;
+
+    @BeforeMethod
+    public void setUp() throws SQLException
+    {
+        db = new TestingDatabase();
+        connection = db.getConnection();
+        connection.createStatement().execute(TABLE_SQL);
+        statement = connection.prepareStatement(INSERT_SQL);
+    }
+
+    @AfterMethod
+    public void tearDown() throws SQLException
+    {
+        db.close();
+    }
+
+    @DataProvider(name = "grid")
+    public static Object[][] rowCountAndBatchSize()
+    {
+        return new Object[][] {
+                {100, 1},
+                {100, 10},
+                {100, 20},
+                {100, 50},
+                {100, 64},
+                {100, 100},
+                {100, 1000},
+                {100, 10000},
+        };
+    }
+
+    @Test(dataProvider = "grid")
+    public void testBatchSizeIsRespectedWhenPageIsAppended(Integer rowsCount, Integer batchSize) throws SQLException
+    {
+        Page page = makePage(rowsCount);
+        JdbcPageSink sink = makeSink(batchSize);
+        sink.appendPage(page);
+
+        int batches = rowsCount / batchSize;
+        int expected = batches * batchSize;
+
+        assertThat(count()).isEqualTo(expected);
+    }
+
+    private int count() throws SQLException
+    {
+        try (Statement st = connection.createStatement(); ResultSet rs = st.executeQuery(COUNT_SQL)) {
```

Comment:
Why not to use `assertQuery(session, query)` like it is used other places. If you write a test using user facing API then test is resilent to changes of internals, and verifies that user facing behaviour is working as expected, this is better than verifying internals.